lib/routes/xiaoheihe/news.ts

```diff
@@ -0,0 +1,64 @@
+import { Route } from '@/types';
+import cache from '@/utils/cache';
+import { load } from 'cheerio';
+import { parseDate } from '@/utils/parse-date';
+import got from '@/utils/got';
+import puppeteer from '@/utils/puppeteer';
+
+export const route: Route = {
+    path: '/news',
+    categories: ['game'],
+    example: '/xiaoheihe/news',
+    features: {
+        requireConfig: false,
+        requirePuppeteer: true,
+        antiCrawler: false,
+        supportBT: false,
+        supportPodcast: false,
+        supportScihub: false,
+    },
+    name: '游戏新闻',
+    maintainers: ['tssujt'],
+    handler,
+};
+
+async function handler() {
+    const browser = await puppeteer({ stealth: true });
+    const response = await got({
+        method: 'get',
+        url: `https://api.xiaoheihe.cn/bbs/app/feeds/news?lang=zh-cn&limit=20&offset=0&tag=-1&version=1.3.303`,
+    });
+    const data = response.data.result.links.filter((item) => item.linkid !== undefined);
+    let items = data.map((item) => ({
+        link: `https://api.xiaoheihe.cn/v3/bbs/app/api/web/share?link_id=${item.linkid}`,
+        title: item.title,
+        pubDate: parseDate(item.modify_at * 1000),
+        description: item.description,
+    }));
+
+    items = await Promise.all(
+        items.map((item) =>
+            cache.tryGet(item.link, async () => {
+                const page = await browser.newPage();
+                await page.goto(item.link, {
+                    waitUntil: 'domcontentloaded',
+                });
+                await page.waitForSelector('.article-content');
+
+                const html = await page.evaluate(() => document.documentElement.innerHTML);
+                await page.close();
+                const $ = load(html);
+                item.description = $('.article-content').html();
+                return item;
```

Comment:
Puppeteer is not really required.
You can use https://github.com/huandu/heybox-url to calculate appropriate `hkey` and `nonce` to obtain the API response at `/bbs/app/api/share/data`

Reply:
I think using Puppeteer will make the availability higher.

> Puppeteer is not really required. You can use [huandu/heybox-url](https://github.com/huandu/heybox-url) to calculate appropriate `hkey` and `nonce` to obtain the API response at `/bbs/app/api/share/data`

Reply:
There are way too many people who don't know they need to deploy Puppeteer or how to deploy it properly. #14812

Reply:
@TonyRL When I try to add [huandu/heybox-url](https://github.com/huandu/heybox-url) package, I get the error `Cannot find module 'heybox-url'`, it seems to be a issue with [huandu/heybox-url](https://github.com/huandu/heybox-url). Is it acceptable if I add some of the source code used in [huandu/heybox-url](https://github.com/huandu/heybox-url) into xiaoheihe/util.ts ?

Reply:
Sure, since it's [MIT](https://github.com/huandu/heybox-url/blob/main/LICENSE) licensed.

Reply:
@TonyRL Updated without Puppeteer. PTAL.